### DISCUSSION TITLE
ESP32 CAM infinite loop after flashing

### DISCUSSION THREAD

**Discussion by fuhraih** [arendst/Tasmota#12136] [Q&A] [answered]
Hello,
I am trying to install Tasmota webcam bin on my new ESP32-CAM board. I am using an ESP32-CAM-MB board in order to flash it with ESP_FLASHER.
When I flash a classic tamsota32-FR.bin. There is no problem, I can access to Tasmota, etc...
When I use the tasmota32-webcam.bin I have logs saying that flashing is done, but after I have infinte loop with message :
[09:21:40]rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[09:21:40]configsip: 0, SPIWP:0xee
[09:21:40]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[09:21:40]mode:QIO, clock div:1
[09:21:40]load:0x3fff0018,len:4
[09:21:40]load:0x91214206,len:575112844
[09:21:40]1162 mmu set 00010000, pos 00010000
[09:21:40]1162 mmu set 00020000, pos 00020000
[09:21:40]1162 mmu set 00030000, pos 00030000
[09:21:40]1162 mmu set 00040000, pos 00040000
[09:21:40]1162 mmu set 00050000, pos 00050000
[09:21:40]1162 mmu set 00060000, pos 00060000
[09:21:40]1162 mmu set 00070000, pos 00070000
[09:21:40]1162 mmu set 00080000, pos 00080000
[09:21:40]1162 mmu set 00090000, pos 00090000
[09:21:40]1162 mmu set 000a0000, pos 000a0000
[09:21:40]1162 mmu set 000b0000, pos 000b0000
[09:21:40]1162 mmu set 000c0000, pos 000c0000
[09:21:40]1162 mmu set 000d0000, pos 000d0000
[09:21:40]1162 mmu set 000e0000, pos 000e0000
[09:21:40]1162 mmu set 000f0000, pos 000f0000
[09:21:40]1162 mmu set 00100000, pos 00100000
[09:21:40]1162 mmu set 00110000, pos 00110000
[09:21:40]ets Jun  8 2016 00:22:57

I tried with release and devlopment versions. All the same.
Does anyone have an idea how I could solve this ?

**Comment by ascillato2** [comment, marked as answer]
Please, try to erase all flash and flash it again directly with the tasmota32-webcam.bin

**Reply by fuhraih** [reply]
ESP_Flasher is supposed to erase flash.
I tried with esptools (erase_flash). The erase worked. But still the same problem when using webcam bin (I even tried with last development version).

**Reply by Jason2866** [reply]
Webcam build uses max. performance settings. Clock 240Mhz. Flash mode QIO and clock 80Mhz. Your board seems not to run stable with this settings, since variant Tasmota does run.

**Reply by sfromis** [reply]
One plausible cause of problems running webcam is if your power source is insufficient. I'd recommend at least 1A of 5V power. Certainly not powered from a USB/serial adapter.

**Comment by cokebottle21** [comment]
I use also a ESP32-Cam, with 9.1.0 all works ok for me. After ota to 9.5.0 its not useable for me. If i remember correct while serial flash with 9.1.0 there was only 40Mhz. I also use ESP_Flasher. So what can i do now to get it work with the newer version?

**Comment by Jason2866** [comment]
@fuhraih @cokebottle21  you can try this version if flash speed 80Mhz is the issue
[webcam_40M_flash.zip](https://github.com/arendst/Tasmota/files/6719855/webcam_40M_flash.zip)
it is build with flash speed 40Mhz. You have to flash via ESP_Flasher since OTA upgrade do NOT change flash mode settings. ESP_Flasher checks the settings and writes the needed settings and bootloader.
Please give me feedback. If it works i will check how big the performance difference is. If the impact is small i will change the default setting for flash clock to 40Mhz.

**Comment by cokebottle21** [comment]
@Jason2866
First, thank you for your very quick reply ;-) My knowledge here is very basic, so I can only report what happens. It is the same for me as with the original 9.5.0, it does not work, just a loop. One thing I also saw is that not only the frequency changed from 40 to 80, but also the flash mode. Could this be another problem with this board? In the past it was dout. In other words maybe 40 Mhz along with dout do the trick?

**Reply by Jason2866** [reply]
The flash mode for webcam build was never dout. V9.1. used mode dio.
This build uses mode dio and 40Mhz for flash
[webcam_40M_flash_dio.zip](https://github.com/arendst/Tasmota/files/6720028/webcam_40M_flash_dio.zip)

**Comment by cokebottle21** [comment]
@Jason2866
>The flash mode for webcam build was never dout

I tried again flashing 9.1.0 there is Flash Mode: dout

Now also tried your new webcam_40M_flash_dio.zip

This one at least boots, but after that it's unusable. It's totally slow, almost no response while I try to configure it. And no webcam stream, only fixed image. I tried to configure template but after reboot it still presents without the configured template and again totally slow. 

After flashing 9.1.0 all worked again as before also the template and all other settings i tried. As i said my knowledge is very basic... i can only test if it works ;-)

**Reply by Jason2866** [reply]
Where do you see it is flash mode dout? I looked in the config files of v9.1 and i see dio.
Maybe i overlook something. Do you can post the whole log file from ESP_FLASHER from beginn of flashing of version 9.1?
Anyway a version with dout
[webcam_40M_flash_dout.zip](https://github.com/arendst/Tasmota/files/6720182/webcam_40M_flash_dout.zip)

**Reply by cokebottle21** [reply]
>Do you can post the whole log

I use ESP-Flasher-Ubuntu 1.1, tasmota32-webcam 9.1.0

Using '/dev/ttyUSB0' as serial port.
Connecting........
Detecting chip type... ESP32
Connecting...

Chip Info:
 - Chip Family: ESP32
 - Chip Model: ESP32-D0WDQ6 (revision 1)
 - Number of Cores: 2
 - Max CPU Frequency: 240MHz
 - Has Bluetooth: YES
 - Has Embedded Flash: NO
 - Has Factory-Calibrated ADC: YES
 - MAC Address: 24:6F:28:xx:xx:xx
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
 - Flash Size: 4MB
 - Flash Mode: dout
 - Flash Frequency: 40MHz
Erasing flash (this may take a while)...
Chip erase completed successfully in 3.5s
Compressed 17104 bytes to 11193...
Writing at 0x00001000... (100 %)
Wrote 17104 bytes (11193 compressed) at 0x00001000 in 0.3 seconds (effective 486.0 kbit/s)...
Hash of data verified.
Compressed 3072 bytes to 129...
Writing at 0x00008000... (100 %)
Wrote 3072 bytes (129 compressed) at 0x00008000 in 0.0 seconds (effective 3033.6 kbit/s)...
Hash of data verified.
Compressed 8192 bytes to 47...
Writing at 0x0000e000... (100 %)
Wrote 8192 bytes (47 compressed) at 0x0000e000 in 0.0 seconds (effective 11331.9 kbit/s)...
Hash of data verified.
Compressed 1214128 bytes to 694701...
Writing at 0x00010000... (2 %)
Writing at 0x00014000... (4 %)
Writing at 0x00018000... (6 %)
Writing at 0x0001c000... (9 %)
Writing at 0x00020000... (11 %)
Writing at 0x00024000... (13 %)
Writing at 0x00028000... (16 %)
Writing at 0x0002c000... (18 %)
Writing at 0x00030000... (20 %)
Writing at 0x00034000... (23 %)
Writing at 0x00038000... (25 %)
Writing at 0x0003c000... (27 %)
Writing at 0x00040000... (30 %)
Writing at 0x00044000... (32 %)
Writing at 0x00048000... (34 %)
Writing at 0x0004c000... (37 %)
Writing at 0x00050000... (39 %)
Writing at 0x00054000... (41 %)
Writing at 0x00058000... (44 %)
Writing at 0x0005c000... (46 %)
Writing at 0x00060000... (48 %)
Writing at 0x00064000... (51 %)
Writing at 0x00068000... (53 %)
Writing at 0x0006c000... (55 %)
Writing at 0x00070000... (58 %)
Writing at 0x00074000... (60 %)
Writing at 0x00078000... (62 %)
Writing at 0x0007c000... (65 %)
Writing at 0x00080000... (67 %)
Writing at 0x00084000... (69 %)
Writing at 0x00088000... (72 %)
Writing at 0x0008c000... (74 %)
Writing at 0x00090000... (76 %)
Writing at 0x00094000... (79 %)
Writing at 0x00098000... (81 %)
Writing at 0x0009c000... (83 %)
Writing at 0x000a0000... (86 %)
Writing at 0x000a4000... (88 %)
Writing at 0x000a8000... (90 %)
Writing at 0x000ac000... (93 %)
Writing at 0x000b0000... (95 %)
Writing at 0x000b4000... (97 %)
Writing at 0x000b8000... (100 %)
Wrote 1214128 bytes (694701 compressed) at 0x00010000 in 20.0 seconds (effective 484.4 kbit/s)...
Hash of data verified.

Leaving...
Hard Resetting...
Done! Flashing is complete!

**Reply by Jason2866** [reply]
I think i found why v9.1 works for you. Probably your board has no PSRAM or a faulty one.
Until this PR https://github.com/espressif/arduino-esp32/pull/4911 platformio didnt support use PSRAM (missing libs). Issue https://community.platformio.org/t/very-bad-performance-if-i-compile-the-example-for-esp32-cam/19730 Tasmota v9.1 is build before this PR was merged. So the Tasmota webcam build v9.1 is a build without supporting PSRAM.
In short boards working with v9.1 and stopped working after March 2021 are crippeled (faulty) ESP32-CAM boards! Without PSRAM higher resolutions and fps are not possible.

**Reply by cokebottle21** [reply]
>I think i found why v9.1 works for you. Probably your board has no PSRAM or a faulty one.

I was now interested what happens if i choose a version between 9.1.0 an 9.5.0

Tried meanwhile now also 9.2.0 and 9.3.1 Both was also as dout and 40 Mhz flashed. It boots, but also not useable. Nearly no response and no Stream only 1 fixed image.

So i can confirm all versions above 9.1.0 with dout and 40 Mhz only boots but not usable, and all versions above 9.1.0 without dout and 80Mhz also dont work because only loops. So iam not sure what you said with PSRAM because 9.2.0 and 9.3.1 was before March 2021? Maybe something else also involved?

**Reply by Jason2866** [reply]
9.2.0 and 9.3.1 are released before March 2021. So the PSRAM fix in Platformio was not included too. I am out of ideas why your board is not working with later versions.
How do you power the board? Powering via the USB adapter does NOT reliable work.
Do you use a decent power supply?

**Reply by cokebottle21** [reply]
I had read that when I started experimenting with it that there may be problems with the power. Therefore, I use since the beginning the USB adapter only for flashing. For booting a separate 2,5A power supply. Also thats tested with a Raspi 3 . The supply Iam sure is fine. Thank you for all your time and help.

**Reply by fuhraih** [reply]
I have tested v9.1 on my card and it works. So the problems comes from my cam.
Thanks for the answers and researchs!

**Reply by Jason2866** [reply]
You can try to do a OTA upgrade to latest version. We have feedback that this worked for many.
